controllers/operator/deployment.go

```diff
@@ -659,7 +704,7 @@ func generateProviderDeploymentObject(instance *operatorv1alpha1.Authentication,
 	return idpDeployment
 }
 
-func generateManagerDeploymentObject(instance *operatorv1alpha1.Authentication, scheme *runtime.Scheme, deployment string, icpConsoleURL string, saasCrnId string) *appsv1.Deployment {
+func generateManagerDeploymentObject(instance *operatorv1alpha1.Authentication, scheme *runtime.Scheme, deployment string, icpConsoleURL string, saasCrnId string, auditSecretExists bool, auditSecretName string) *appsv1.Deployment {
```

Comment:
What I've said RE: using the `auditSecretName` as the signal for whether the Secret exists applies here, too - you should be able to just check to see if the name is empty and then request the audit tls volumes when it isn't.